plotAfb.py

```diff
@@ -0,0 +1,287 @@
+import argparse	
+import ROOT
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+
+from ROOT import TCanvas, TPad, TH1F, TH1I, THStack, TLegend, TMath, gROOT
+import ratios
+from setTDRStyle import setTDRStyle
+gROOT.SetBatch(True)
+from helpers import *
+from defs import getPlot, Backgrounds, Backgrounds2016, Backgrounds2018, Signals, Signals2016, Signals2016ADD, Data, Data2016, Data2018, path, plotList, zScale, zScale2016, zScale2018, defineMyColors
+import math
+import os
+from copy import copy
+def plotDataMC(plotsPos,plotsNeg, background):
+	#hCanvas = TCanvas("hCanvas", "Distribution", 800,800)
+	#plotPad = ROOT.TPad("plotPad","plotPad",0,0,1,1)
+	#setTDRStyle()
+	#plotPad.UseCurrentStyle()
+	#plotPad.Draw()
+	#plotPad.cd()
+	#colors = createMyColors()
+	
+
+
+	histTree={}
+	lumi={}
+	lumi["muon"]={"2016":36.3*1000, "2017":42.135*1000, "2018":61.608*100}
+	lumi["electron"]={"2016":35.9*1000,"2017":41.529*1000,"2018":59.97*1000}
+	dataTree={"2016":Process(Data2016, normalized=True),"2017":Process(Data, normalized=True),"2018":Process(Data2018, normalized=True)}
+	backgroundDict={"2016":Backgrounds2016,"2017":Backgrounds,"2018":Backgrounds2018}
+	backgroundTree={}
+	eventCounts={}
+	eventCounts["electron"]=totalNumberOfGeneratedEvents(path,False)
+	eventCounts["muon"]=totalNumberOfGeneratedEvents(path)
+	negWeights={}
+	negWeights["electron"]=negWeightFractions(path,False)
+	negWeights["muon"]=negWeightFractions(path)
+	
+	process2016=[]
+	process2017=[]
+	process2018=[]
+	for flavor in ["muon","electron"]:
+		backgroundTree[flavor]={}
+		for year in ["2016","2017","2018"]:
+			backgroundTree[flavor][year]=[]
+			for background in backgrounds:
+				backgroundTree[flavor][year].append(Process(getattr(backgroundDict[year],background),eventCounts[flavor],negWeights[flavor]))	
+	zScaleFac={}
+	zScaleFac["muon"]={"2016":zScale2016["muons"],"2017":zScale["muons"],"2018":zScale2018["muons"]}
+	zScaleFac["electron"]={"2016":zScale2016["electrons"],"2017":zScale["electrons"],"2018":zScale2018["electrons"]}
+	print(zScaleFac)
+	plotTree={}
+	for flavor in ["muon","electron"]:
+		plotTree[flavor]={}
+	plotTree["muon"]["BB"]=[plotsPos[0],plotsNeg[0]]
+	plotTree["muon"]["BE"]=[plotsPos[1],plotsNeg[1]]
+	plotTree["electron"]["BB"]=[plotsPos[2],plotsNeg[2]]
+	plotTree["electron"]["BE"]=[[plotsPos[3],plotsNeg[3]],[plotsPos[4],plotsNeg[4]]]	
+	for flavor in ["muon","electron"]:
+		bng=binning(flavor)
```

Comment:
The binning is way too fine for this plot. Please define a more suitable binning for the AFB plots.